pkg/didmethod/peer/did.go

```diff
@@ -30,7 +32,59 @@ func newDid(doc *did.Doc) (string, error) {
 	if doc.PublicKey == nil || doc.Authentication == nil {
 		return "", errors.New("the genesis version must include public keys and authentication")
 	}
+	numBasis, err := hashString(doc)
+	if err != nil {
+		return "", err
+	}
+	messageIdentifier := []string{peerPrefix, numAlgo, encAlgo, "-", numBasis}
+	peerDID := strings.Join(messageIdentifier, "")
+
+	return peerDID, nil
+}
+
+func validate(doc *did.Doc) error {
+	did := doc.ID
+	hashedNumBasis, err := validateDid(did)
+	if err != nil {
+		return err
+	}
+	//removing did to calculate the hash of genesis doc
+	doc.ID = ""
```

Comment:
It's a bad idea to mutate input args on validation functions. Make a copy of the input doc instead.

Reply:
created a copy of the input doc